gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/InMemoryDataStore.js

```diff
@@ -223,29 +243,37 @@ var fluid = fluid || require("infusion");
         return authDecision;
     };
 
-    gpii.oauth2.dataStore.updateAuthDecision = function (authDecisions, applier, userId, authDecisionData) {
+    gpii.oauth2.dataStore.updateAuthDecision = function (authDecisions, gpiiTokens, applier, userId, authDecisionData) {
         // Only update the authDecision if it is owned by userId so that
         // users cannot update authorizations owned by others
         var authDecision = gpii.oauth2.dataStore.findAuthDecisionById(authDecisions, authDecisionData.id);
-        if (authDecision && authDecision.userId === userId) {
-            authDecision.selectedPreferences = fluid.copy(authDecisionData.selectedPreferences);
-            // We are changing one of the authDecision items in the collection but
-            // notifying on the collection (it's not worth it to extend the query api
-            // to allow a more fine grained update)
-            applier.change("authDecisions", authDecisions);
+        if (authDecision) {
+            var gpiiToken = gpii.oauth2.dataStore.findGpiiToken(gpiiTokens, authDecision.gpiiToken);
+            if (gpiiToken && gpiiToken.userId === userId) {
+                authDecision.selectedPreferences = fluid.copy(authDecisionData.selectedPreferences);
+                // We are changing one of the authDecision items in
+                // the collection but notifying on the collection
+                // (it's not worth it to extend the query api to allow
+                // a more fine grained update)
+                applier.change("authDecisions", authDecisions);
+            }
         }
     };
 
-    gpii.oauth2.dataStore.revokeAuthDecision = function (authDecisions, applier, userId, authDecisionId) {
+    gpii.oauth2.dataStore.revokeAuthDecision = function (authDecisions, gpiiTokens, applier, userId, authDecisionId) {
         // Only revoke the authorization with authDecisionId if it is owned
         // by userId so that users cannot revoke authorizations owned by others
         var authDecision = gpii.oauth2.dataStore.findAuthDecisionById(authDecisions, authDecisionId);
-        if (authDecision && authDecision.userId === userId) {
-            authDecision.revoked = true;
-            // We are changing one of the authDecision items in the collection but
-            // notifying on the collection (it's not worth it to extend the query api
-            // to allow a more fine grained update)
-            applier.change("authDecisions", authDecisions);
+        if (authDecision) {
+            var gpiiToken = gpii.oauth2.dataStore.findGpiiToken(gpiiTokens, authDecision.gpiiToken);
+            if (gpiiToken && gpiiToken.userId === userId) {
+                authDecision.revoked = true;
+                // We are changing one of the authDecision items in
```

Comment:
All these comments have been refactored to pretty short lines (shorter than the code surrounding them), some kind of automated IDE operation got out of hand?